### ISSUE TITLE
path management in windows

### ISSUE THREAD

**Issue by boehlim377** [lsnepomuceno/laravel-a1-pdf-sign#161]
If the project is in a folder that contains a space, the openssl command will not work. Because then the Temp folder cannot be used.
If the certificate to be used has a space (e.g. in the path), it won't work either

The problem occurs in the file \src\Sign\ManageCert.php in Line 81
`$openSslCommand = "openssl pkcs12 -in {$pfxPath} -out {$output} -nodes -password pass:{$shellArgPassword} {$legacyFlag}";`

In the a1TempDir function the slashes should be replaceced with DIRECTORY_SEPARATOR. (Helpers\helpers.php line 109 and 112)

**To Reproduce**
1. Create an project in Windows (10) with "C:\projets\my Project\"
2.  $cert = new ManageCert;
3. $cert->fromPfx("C:\projets\my Project\test.pfx", password)

**Expected behavior**
All used paths (certificate, temp) should be escaped.

**Versions involved:**
- Laravel: [10.48.22]
- lsnepomuceno/laravel-a1-pdf-sign [1.3]

**possible fix**
```
$openSslCommand = "openssl pkcs12 -in " . escapeshellarg($pfxPath) . " -out " . escapeshellarg($output) . " -nodes -password pass:{$shellArgPassword} {$legacyFlag}";
```

**Comment by cpibia** [comment]
The TCPDF documentation uses `file://` to indicate an absolute path to the certificate file. In your case, it looks like you are using a Windows path.

Try changing the certificate path to this:

```php
$certificate = 'file:///C:/projets/my Project/test.pfx'; 
```

`file://`: This tells TCPDF that you are providing an absolute path to the file.
`/C:/projets/my Project/test.pfx`: This is the absolute path to your .pfx file on Windows, with slashes (/) instead of backslashes ().
**Important**: Make sure that the test.pfx file actually exists in the specified path and that TCPDF has permissions to read it.

in any case, i suggest you not to work with Windows but to use a Linux machine for PHP development. WSL2 could be a solution.

Regards
Cristian

**Comment by lsnepomuceno** [comment]
Hello @boehlim377,

Firstly, I would like to thank you for the feedback regarding the file path problem when extracting data from the pfx file.

The problem has been fixed and is now available from [v1.3.1](https://github.com/lsnepomuceno/laravel-a1-pdf-sign/releases/tag/1.3.1)
